src/swe/swe_ceed.c

```diff
@@ -485,41 +485,80 @@ PetscErrorCode SWESourceOperatorSetTimeStep(CeedOperator source_op, PetscReal dt
 }
 
 // Given a shallow water equations source operator created by
-// CreateSWESourceOperator, fetches the field representing the source of water.
+// CreateSWESourceOperator, fetches the field associated with the field_name.
 // This can be used to implement a time-dependent water source.
-PetscErrorCode SWESourceOperatorGetWaterSource(CeedOperator source_op, CeedOperatorField *water_source_field) {
+static PetscErrorCode SWESubOperatorGetOperatorField(CeedOperator op, CeedInt sub_op_idx, const char *field_name, CeedOperatorField *op_field) {
   PetscFunctionBeginUser;
 
   // get the source sub-operator responsible for the water source (the first one)
   CeedOperator *sub_ops;
-  CeedCompositeOperatorGetSubList(source_op, &sub_ops);
-  CeedOperator water_source_op = sub_ops[0];
+  CeedCompositeOperatorGetSubList(op, &sub_ops);
+  CeedOperator sub_op = sub_ops[sub_op_idx];
```

Comment:
@jedbrown The code here extracts the `sub_op_idx`-th Ceed suboperator. I want to add a check earlier to ensure the Ceed operator has the `sub_op_idx` suboperator via

```
  CeedInt num_sub_op;
  CeedCompositeOperatorGetNumSub(op, &num_sub_op);
 
// Now Assert sub_op_idx < num_sub_op

  // get the source sub-operator responsible for the water source (the first one)
  CeedOperator *sub_ops;
  CeedCompositeOperatorGetSubList(op, &sub_ops);
  CeedOperator sub_op = sub_ops[sub_op_idx];
```
Is there a `CeedAssert` equivalent to `PetscAssert`? Or, should I directly use `exit()`?

Reply:
I think I can use `PetscCheck`.